client/consensus/slots/src/lib.rs

```diff
@@ -778,15 +812,15 @@ mod test {
 	use super::*;
 	use sp_runtime::traits::NumberFor;
 	use std::time::{Duration, Instant};
-	use substrate_test_runtime_client::runtime::{Block, Header};
+	use substrate_test_runtime_client::runtime::{Block, Header, TestInherentDataProvider};
 
 	const SLOT_DURATION: Duration = Duration::from_millis(6000);
 
 	fn slot(slot: u64) -> super::slots::SlotInfo<Block> {
 		super::slots::SlotInfo {
 			slot: slot.into(),
 			duration: SLOT_DURATION,
-			inherent_data: Default::default(),
+			create_inherent_data: Box::new(TestInherentDataProvider),
```

Comment:
```suggestion
			create_inherent_data: Box::new(|_, ()| Ok(sp_timestamp::InherentDataProvider::from_system_time())),
```

This should make this compile without requiring the `TestInherentDataProvider`.